AutoRest/Generators/Azure.Common/Azure.Common/AzureCodeGenerator.cs

```diff
@@ -204,6 +207,84 @@ public static void AddLongRunningOperations(ServiceClient serviceClient)
         }
 
         /// <summary>
+        /// Adds the parameter groups to operation parameters.
+        /// </summary>
+        /// <param name="serviceClient"></param>
+        public static void AddParameterGroups(ServiceClient serviceClient)
+        {
+            if (serviceClient == null)
+            {
+                throw new ArgumentNullException("serviceClient");
+            }
+
+            foreach (Method method in serviceClient.Methods)
+            {
+                //TODO: Does this group name need to be normalized later, or does that already happen?
+                Dictionary<string, List<Property>> properties = new Dictionary<string, List<Property>>();
+                foreach (Parameter parameter in method.Parameters)
+                {
+                    if (parameter.Extensions.ContainsKey(ParameterGroupExtension))
+                    {
+                        Newtonsoft.Json.Linq.JContainer extensionObject = parameter.Extensions[ParameterGroupExtension] as Newtonsoft.Json.Linq.JContainer;
+                        if (extensionObject != null)
+                        {
+                            string parameterGroupName = method.Group + "-" + method.Name + "-" + "Parameters";
+                            parameterGroupName = extensionObject.Value<string>("name") ?? parameterGroupName;
+                            
+                            parameter.ParameterGroup = parameterGroupName;
+
+                            if (!properties.ContainsKey(parameterGroupName))
+                            {
+                                properties.Add(parameterGroupName, new List<Property>());
+                            }
+
+                            properties[parameterGroupName].Add(new Property()
+                                {
+                                    IsReadOnly = false, //Since these properties are used as parameters they are never read only
+                                    Name = CodeNamer.PascalCase(parameter.Name),
```

Comment:
any reason, you like to do the pascal conversion here? If you don't do it would c# generator produces property name with wrong casing?